Makefile

```diff
@@ -8,35 +8,30 @@ mypy:
 	mypy --ignore-missing-imports $(MODULES)
 
 test_srcs := $(wildcard tests/test_*.py)
-all_test_srcs := $(addprefix all__, $(test_srcs))
-integration_test_srcs := $(addprefix integration__, $(test_srcs))
 
-test: lint mypy $(test_srcs)
+define test_rules
+$1_test_srcs := $$(addprefix $1__, $$(test_srcs))
+$1_test: lint mypy $$($1_test_srcs)
 	coverage combine
 	rm -f .coverage.*
 
-$(test_srcs): %.py :
-	set -o pipefail \
-	&& DSS_TEST_MODE="standalone" coverage run -p --source=dss -m unittest $(DSS_UNITTEST_OPTS) $@  2>&1 \
-	| sed -e "s/^/$$$$ /"
+$$($1_test_srcs): $1__%.py :
+	@set -o pipefail \
+	&& (DSS_TEST_MODE="$2" coverage run -p --source=dss -m unittest $$(DSS_UNITTEST_OPTS) $$*.py) 2>&1 \
+	| sed -e "s/^/$$$$$$$$ /"
 
-integration_test: lint mypy $(integration_test_srcs)
-	coverage combine
-	rm -f .coverage.*
+.PHONY: $1_test $$($1_test_srcs)
 
-$(integration_test_srcs): integration__%.py :
-	set -o pipefail \
-	&& DSS_TEST_MODE="integration" coverage run -p --source=dss -m unittest $(DSS_UNITTEST_OPTS) $*.py 2>&1 \
-	| sed -e "s/^/$$$$ /"
+# Serialize the tests that use a local Elasticsearch instance
+$1__tests/test_search.py: $1__tests/test_indexer.py
```

Comment:
Correct me if I'm wrong, but does this not mean that if you run `test_search` from the command line, this will also run test_indexer?

Reply:
You're right. I should use order-only dependencies here. Will fix.

Reply:
Order-only dependencies won't fix this.

Reply:
We could, I guess, switch back to the system where you are *compelled* to have a running ES instance, and have `.travis.xml` kick that off.

Reply:
That's worth considering. I would like to propose that we try the heap reduction for a few days and then explore your proposal in a separate PR.